packages/google_sign_in/android/src/main/java/io/flutter/plugins/googlesignin/GoogleSignInPlugin.java

```diff
@@ -245,39 +240,27 @@ public void getTokens(final Result result, final String email) {
         return;
       }
 
-      Callable<String> getTokenTask =
-          new Callable<String>() {
-            @Override
-            public String call() throws Exception {
-              Account account = new Account(email, "com.google");
-              String scopesStr = "oauth2:" + Joiner.on(' ').join(requestedScopes);
-              return GoogleAuthUtil.getToken(registrar.context(), account, scopesStr);
-            }
-          };
-
-      backgroundTaskRunner.runInBackground(
-          getTokenTask,
-          new BackgroundTaskRunner.Callback<String>() {
-            @Override
-            public void run(Future<String> tokenFuture) {
-              try {
-                String token = tokenFuture.get();
-                HashMap<String, String> tokenResult = new HashMap<>();
-                tokenResult.put("accessToken", token);
-                // TODO(jackson): If we had a way to get the current user at this
-                // point, we could use that to obtain an up-to-date idToken here
-                // instead of the value we cached during sign in. At least, that's
-                // how it works on iOS.
-                result.success(tokenResult);
-              } catch (ExecutionException e) {
-                Log.e(TAG, "Exception getting access token", e);
-                result.error(ERROR_REASON_EXCEPTION, e.getCause().getMessage(), null);
-              } catch (InterruptedException e) {
-                result.error(ERROR_REASON_EXCEPTION, e.getMessage(), null);
-                Thread.currentThread().interrupt();
-              }
-            }
-          });
+      GetOauthTokenTask task =
+          new GetOauthTokenTask(
+              new GetOauthTokenTask.OnTokenListener() {
+                @Override
+                public void onToken(@NonNull String token) {
+                  HashMap<String, String> tokenResult = new HashMap<>();
+                  tokenResult.put("accessToken", token);
+                  // TODO(jackson): If we had a way to get the current user at this
+                  // point, we could use that to obtain an up-to-date idToken here
+                  // instead of the value we cached during sign in. At least, that's
+                  // how it works on iOS.
+                  result.success(tokenResult);
+                }
+
+                @Override
+                public void onError(@NonNull Throwable error) {
+                  Log.e(TAG, "Exception getting access token", error);
+                  result.error(ERROR_REASON_EXCEPTION, error.getCause().getMessage(), null);
```

Comment:
This results in a `NullPointerException`, if the `error` has no cause. In the old code, the error we were interested in would always be wrapped in an `ExecutionException`. Not so here, I think?